### ISSUE TITLE
Missing one extension while generating JA3 fingerprint

### ISSUE THREAD

**Issue by amitguptakiitian** [seladb/PcapPlusPlus#689]
Hi,
I observed there is a difference in JA3 fingerprint/hash generated by pcapplusplus and JA3 fingerprint generated by a python tool (ja3) available in ubuntu.
for my experiment, I dump the pcap file on my router and used this python tool (ja3) to extract the fingerprint and the same packet is feed to pcapplusplus as well to extract the ja3 fingerprint.

Below is the ja3 generated on ubuntu using ja3 python tool.
"ja3": "771,52392-52393-49199-49200-49195-49196-49171-49161-49172-49162-156-157-47-53-49170-10,0-5-10-11-13-65281-18,29-23-24-25,0",
and below is the fingerprint received from pcapplusplus.
771,52392-52393-49199-49200-49195-49196-49171-49161-49172-49162-156-157-47-53-49170-10,0-5-10-11-13-65281,29-23-24-25,0

**Here 18 is missing after 65281**. can you please confirm and let us know whether it's a bug or am I doing something wrong?
thank you in advance.
Best Regards
Amit Kumar.

**Comment by seladb** [comment]
hi @amitguptakiitian can you attach a packet that reproduces this issue? I can look into it

**Comment by amitguptakiitian** [comment]
sure, attached the pcap file.
[test_ja3_out.zip](https://github.com/seladb/PcapPlusPlus/files/6967701/test_ja3_out.zip)

**Comment by amitguptakiitian** [comment]
Hi @seladb , please let us know if you need more info to reproduce this issue.
thanks
Amit

**Comment by seladb** [comment]
hi @amitguptakiitian , sorry for the delayed response. I found a bug in the way TLS client-hello packets are parsed - the code wasn't parsing TLS extensions with zero size properly. That's why it ignored the last extension in the packet you attached.
I opened a PR with the fix: https://github.com/seladb/PcapPlusPlus/pull/691
I'll soon merge it to `dev` and `master`.

With this fix the TLS fingerprinting should work as expected.

Thanks for reporting this issue!
